custom_components/omnik_inverter/sensor.py

```diff
@@ -128,3 +249,41 @@ def native_value(self) -> StateType:
         if isinstance(value, str):
             return value.lower()
         return value
+
+
+class OmnikInverterArraySensorEntity(OmnikInverterSensorEntity):
+    """Defines an Omnik Inverter sensor reading from an array."""
+
+    def __init__(
+        self,
+        *,
+        coordinator: OmnikInverterDataUpdateCoordinator,
+        index: int,
+        description: ArraySensorEntityDescription,
+        service_key: Literal["inverter", "device"],
+        name: str,
+        service: str,
+    ) -> None:
+        self.index = index
+        self.data_key = description.data_key
+        human_index = index + 1
+
+        description = dataclasses.replace(
+            description,
+            key=description.key.format(human_index),
+            name=description.name.format(human_index),
+        )
+
+        super().__init__(
+            coordinator=coordinator,
+            description=description,
+            service_key=service_key,
+            name=name,
+            service=service,
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        value = getattr(self.coordinator.data[self._service_key], self.data_key)
+        return value[self.index]
```

Comment:
Note there's no `None`-propagation here so this'll crash if not running on the TCP backend as the array should be `None` instead of `[None, None, None]`. I'll fix that on the next revision.